src/transformers/models/mvp/modeling_mvp.py

```diff
@@ -0,0 +1,2055 @@
+            else:
+
+                layer_outputs = decoder_layer(
+                    hidden_states,
+                    attention_mask=attention_mask,
+                    encoder_hidden_states=encoder_hidden_states,
+                    encoder_attention_mask=encoder_attention_mask,
+                    layer_head_mask=(head_mask[idx] if head_mask is not None else None),
+                    cross_attn_layer_head_mask=(
+                        cross_attn_head_mask[idx] if cross_attn_head_mask is not None else None
+                    ),
+                    self_attn_prompt=(self_attn_prompt[idx] if self.use_prompt else None),
+                    cross_attn_prompt=(cross_attn_prompt[idx] if self.use_prompt else None),
+                    past_key_value=past_key_value,
+                    output_attentions=output_attentions,
+                    use_cache=use_cache,
+                )
+            hidden_states = layer_outputs[0]
+
+            if use_cache:
+                next_decoder_cache += (layer_outputs[3 if output_attentions else 1],)
+
+            if output_attentions:
+                all_self_attns += (layer_outputs[1],)
+
+                if encoder_hidden_states is not None:
+                    all_cross_attentions += (layer_outputs[2],)
+
+        # add hidden states from the last decoder layer
+        if output_hidden_states:
+            all_hidden_states += (hidden_states,)
+
+        next_cache = next_decoder_cache if use_cache else None
+        if not return_dict:
+            return tuple(
+                v
+                for v in [hidden_states, next_cache, all_hidden_states, all_self_attns, all_cross_attentions]
+                if v is not None
+            )
+        return BaseModelOutputWithPastAndCrossAttentions(
+            last_hidden_state=hidden_states,
+            past_key_values=next_cache,
+            hidden_states=all_hidden_states,
+            attentions=all_self_attns,
+            cross_attentions=all_cross_attentions,
+        )
+
+
+@add_start_docstrings(
+    "The bare MVP Model outputting raw hidden-states without any specific head on top.",
+    MVP_START_DOCSTRING,
+)
+class MvpModel(MvpPretrainedModel):
+    _keys_to_ignore_on_load_unexpected = [r"final_logits_bias", r"lm_head.weight"]
+
+    def __init__(self, config: MvpConfig):
+        super().__init__(config)
+
+        padding_idx, vocab_size = config.pad_token_id, config.vocab_size
+        self.shared = nn.Embedding(vocab_size, config.d_model, padding_idx)
+
+        self.encoder = MvpEncoder(config, self.shared, config.use_prompt)
+        self.decoder = MvpDecoder(config, self.shared, config.use_prompt)
+
+        # Whether to use lightweight tuning
+        if config.lightweight_tuning:
+            self.requires_grad_(False)
+            self.encoder.self_attn_prompt.requires_grad_(True)
+            self.decoder.self_attn_prompt.requires_grad_(True)
+            self.decoder.cross_attn_prompt.requires_grad_(True)
```

Comment:
Looks like something that should be set in a method instead of a config parameter (like we do for gradient checkpointing). If the config saved on the hub as this option `True`, then a user will automatically get some parameters with no gradients which will introduce hard-to-debug behavior.

So instead of `config.lightweight_tuning` we shoujld have a `MvpPreTrainedModel.set_lightweight_tuning` method.

Reply:
I think it is a great idea! thx!